From bc0fe89315ac2f2a7a68d2d19df9a6a17a575fee Mon Sep 17 00:00:00 2001
From: Pavel Hrdina <phrdina@redhat.com>
Date: Wed, 10 May 2017 10:00:39 +0200
Subject: interface: don't print error for active interface without an IP
 address

If there is no IP address for active interface we need to skip
probing it.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1449509

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
(cherry picked from commit 86e2c23d71a1bdaa153a578e98bf91d47a045f36)
Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
---
 virtManager/interface.py | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/virtManager/interface.py b/virtManager/interface.py
index 2fe61b2a..ce1c1eba 100644
--- a/virtManager/interface.py
+++ b/virtManager/interface.py
@@ -138,13 +138,14 @@ class vmmInterface(vmmLibvirtObject):
             return None, []
 
         ret = []
-        for ip in activeProto.ips:
-            ipstr = ip.address
-            if not ipstr:
-                continue
-            if ip.prefix:
-                ipstr += "/%s" % ip.prefix
-            ret.append(ipstr)
+        if activeProto:
+            for ip in activeProto.ips:
+                ipstr = ip.address
+                if not ipstr:
+                    continue
+                if ip.prefix:
+                    ipstr += "/%s" % ip.prefix
+                ret.append(ipstr)
         return inactiveProto or activeProto, ret
 
     def get_ipv4(self):
-- 
2.13.0

